lib/manager/bundler/artifacts.ts

```diff
@@ -125,11 +141,19 @@ export async function updateArtifacts(
       `gem install bundler${bundlerVersion} --no-document`,
     ];
 
+    const bundlerHostRulesVariables = findAll({ hostType: 'bundler' })
+      .filter(
+        hostRule =>
+          hostRule.domainName && hostRule.username && hostRule.password
+      )
+      .map(hostRule => buildBundleHostVariable(hostRule));
+
     const execOptions: ExecOptions = {
       docker: {
         image: 'renovate/ruby',
         tag: await getDockerTag(updateArtifact),
         preCommands,
+        envVars: bundlerHostRulesVariables,
```

Comment:
You are thinking of setting up the env variable for the child process when the child is not a docker run invocation right ? I think we should be able to support that fairly easy.

Reply:
Yes, that's what I'm thinking. As they are passed via env (instead of `bundle config` commands) then it should be only a sandboxed change that doesn't "affect" the user's original env once Renovate finishes.

Reply:
Now we are passing the envVariables as `extraEnv` to `exec` and that will add the variables to the `childEnv`.